code/ATMOSPHERICS/components/components_base.dm

```diff
@@ -199,7 +195,6 @@ Helpers
 */
 
 /obj/machinery/atmospherics/components/proc/update_parents()
-	for(var/P in parents)
-		var/datum/pipeline/parent = parents["[P]"]
-		if (parent && istype(parent))
-			parent.update = 1
\ No newline at end of file
+	for(var/I = 1; I <= device_type; I++)
+		var/datum/pipeline/parent = parents["p[I]"]
+		parent.update = 1
```

Comment:
Not sure why you changed this considering for(var/P in parents) is twice as fast as for(var/I = 1; I <= device_type; I==)

Course it probably doesn't matter either, it's just a weird change